src/vs/base/common/comparers.ts

```diff
@@ -6,30 +6,168 @@
+	const collator = new Intl.Collator(undefined, { numeric: true });
+	return {
+		collator: collator
+	};
+});
+
+// A collator with numeric sorting enabled, and sensitivity to accents but not case.
+const intlFileNameCollatorNumericCaseInsenstive: IdleValue<{ collator: Intl.Collator }> = new IdleValue(() => {
+	const collator = new Intl.Collator(undefined, { numeric: true, sensitivity: 'accent' });
 	return {
-		collator: collator,
-		collatorIsNumeric: collator.resolvedOptions().numeric
+		collator: collator
 	};
 });
 
-export function compareFileNames(one: string | null, other: string | null, caseSensitive = false): number {
-	const a = one || '';
-	const b = other || '';
-	const result = intlFileNameCollator.getValue().collator.compare(a, b);
+// A collator with numeric sorting disabled.
+const intlFileNameCollator: IdleValue<{ collator: Intl.Collator }> = new IdleValue(() => {
+	const collator = new Intl.Collator(undefined, { numeric: false });
+	return {
+		collator: collator
+	};
+});
 
-	// Using the numeric option in the collator will
-	// make compare(`foo1`, `foo01`) === 0. We must disambiguate.
-	if (intlFileNameCollator.getValue().collatorIsNumeric && result === 0 && a !== b) {
-		return a < b ? -1 : 1;
+// A collabor with numeric sorting disabled and sensitivity to accents but not case.
+const intlFileNameCollatorCaseInsensitive: IdleValue<{ collator: Intl.Collator }> = new IdleValue(() => {
+	const collator = new Intl.Collator(undefined, { numeric: false, sensitivity: 'accent' });
+	return {
+		collator: collator
+	};
+});
+
+// Export compareFileNamesNumeric by its old name too for backwards compatibility.
+// Also keeping a placeholder parameter for the same reason. The parameter was sometimes supplied but was
+// not used in the original code.
+export { compareFileNamesNumeric as compareFileNames };
+
+/** Compares filenames by name then extension, sorting numbers numerically instead of alphabetically. */
+export function compareFileNamesNumeric(one: string | null, other: string | null, placeholder = false): number {
+	const [oneName, oneExtension] = extractNameAndExtension(one);
+	const [otherName, otherExtension] = extractNameAndExtension(other);
+	const collatorNumeric = intlFileNameCollatorNumeric.getValue().collator;
+	const collatorNumericCaseInsensitive = intlFileNameCollatorNumericCaseInsenstive.getValue().collator;
+	let result;
+
+	// Check for name differences, comparing numbers numerically instead of alphabetically.
+	result = collatorNumeric.compare(oneName, otherName);
+	if (result !== 0) {
+		return result;
 	}
 
-	return result;
+	// Using the numeric option in the collator will make compare(`foo1`, `foo01`) === 0. Sort the shorter name first.
+	if (oneName.length !== otherName.length) {
+		return oneName.length < otherName.length ? -1 : 1;
+	}
+
+	// Check for case insensitive extension differences, comparing numbers numerically instead of alphabetically.
```

Comment:
Extensions are a case where it seems best to use a case-insensitive (but not accent-insensitive) check before disambiguating numbers. Their case only comes into play at the very end if everything else has no difference. This makes sense because `.MD` and `.md` are treated as the same thing when determining file type.